core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -83,7 +83,7 @@ object TransactionCoordinator {
 class TransactionCoordinator(brokerId: Int,
                              txnConfig: TransactionConfig,
                              scheduler: Scheduler,
-                             producerIdManager: ProducerIdManager,
+                             producerIdGeneratorFactory: () => ProducerIdGenerator,
```

Comment:
Nit: since this is a function, I'd call it `createProduceIdGenerator` or a verb like that.